go.mod

```diff
@@ -3,37 +3,61 @@ module github.com/kcp-dev/code-generator
 go 1.18
 
 require (
+	github.com/kcp-dev/apimachinery v0.0.0-20220511165638-c591795618c7
+	github.com/kcp-dev/logicalcluster v1.0.0
 	github.com/onsi/ginkgo v1.16.5
 	github.com/onsi/gomega v1.17.0
 	github.com/spf13/cobra v1.2.1
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/mod v0.4.2
-	golang.org/x/tools v0.1.6-0.20210820212750-d4cc65f0b2ff
-	k8s.io/code-generator v0.23.6
+	golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3
+	golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
+	k8s.io/apimachinery v0.24.0
+	k8s.io/client-go v0.24.0
```

Comment:
do we want to be pulling the 1.24 libraries or stick with 1.23 like the rest of the project

Reply:
I can go back to 1.23 to be on the safe side

Reply:
Fixed